### PR TITLE
Add docs and scripts for Azure -> GCP workload identity federation

### PR DESCRIPTION
**Pull request by asatwal**
## Context
Add scripts and docs for workload identity federation. Trello card [here](https://trello.com/c/nbML53b1).

## Changes proposed in this pull request
- Scripts for creating a workload identity pool and provider on GCP
- Basic documentation 

## Guidance to review
- Requires gcloud CLI and login for testing

## Checklist

- [x] I have performed a self-review of my code, including formatting and typos
- [x] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [x] I have added the `Devops` label
- [x] I have attached the pull request to the trello card

### CODE REVIEW

File: scripts/azure-gcp-wif/create-gcp-workload-identity-pool-provider.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# - gcloud init
+# - gcloud auth login
+PROJECT="rugged-abacus-218110"
+
+# Create workload identity pool
+# NOTE: must be "global" region (Can't use for example "europe-west2")
+gcloud iam workload-identity-pools providers create-oidc azure-cip-oidc-provider \
+ --location="global" \
+ --workload-identity-pool="azure-cip-identity-pool" \
+ --issuer-uri="https://login.microsoftonline.com/9c7d9dd3-840c-4b3f-818e-552865082e16/v2.0" \
```
Comment (saliceti) [inline scripts/azure-gcp-wif/create-gcp-workload-identity-pool-provider.sh:13]: 9c7d9dd3-840c-4b3f-818e-552865082e16 is the azure tenant id. can you add it as a variable (like $PROJECT)

File: scripts/azure-gcp-wif/create-gcp-workload-identity-pool-provider.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ --allowed-audiences="fb60f99c-7a34-4190-8149-302f77469936" \
```
Comment (saliceti) [inline scripts/azure-gcp-wif/create-gcp-workload-identity-pool-provider.sh:14]: fb60f99c-7a34-4190-8149-302f77469936 is the AzureADTokenExchange app id (I think!). can you add it as a variable (like $PROJECT)

File: scripts/azure-gcp-wif/create-gcp-workload-identity-pool-provider.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
```
Comment (saliceti) [inline scripts/azure-gcp-wif/create-gcp-workload-identity-pool-provider.sh:1]: We normally use `#!/usr/bin/env sh` whenever possible as it's more portable

File: scripts/azure-gcp-wif/create-gcp-workload-identity-pool.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```
Comment (saliceti) [inline scripts/azure-gcp-wif/create-gcp-workload-identity-pool.sh:1]: We normally use `#!/usr/bin/env sh` whenever possible as it's more portable

File: documentation/azure-gcp-workload-identity-federation.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Azure GCP Workload Identity Federation
+
+This allows Azure AKS client to access GCP resources without the use of plain text API Keys, that are considered a security risk.
+
+Workload Identity Federation requires configuration on both Azure and GCP.
+
+## GCP Configuration
```
Comment (saliceti) [inline documentation/azure-gcp-workload-identity-federation.md:7]: How do you authenticate?
Which level of permission do you need?
How to get a GCP account?

File: documentation/azure-gcp-workload-identity-federation.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+### Create a workload identity pool
+
+GCloud bash script:
+
+```
+create-gcp-workload-identity-pool.sh
```
Comment (saliceti) [inline documentation/azure-gcp-workload-identity-federation.md:14]: Can we add the full path (script/azure-gcp-wif/xxx)

File: documentation/azure-gcp-workload-identity-federation.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+```
+
+### Create a workload identity pool provider
+
+GCloud bash script:
+
+```
+create-gcp-workload-identity-pool-provider.sh
```
Comment (saliceti) [inline documentation/azure-gcp-workload-identity-federation.md:22]: Can we add the full path (scripts/azure-gcp-wif/xxx)

File: documentation/azure-gcp-workload-identity-federation.md (hunk 5 above)
Comment (asatwal) [reply documentation/azure-gcp-workload-identity-federation.md:7]: Now added.

File: scripts/azure-gcp-wif/create-gcp-workload-identity-pool-provider.sh [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env sh
+
+# Requires gcloud cli and authentication with gcloud:
+# - gcloud init
+# - gcloud auth login
+PROJECT="rugged-abacus-218110"
+AZURE_TENANT_ID="9c7d9dd3-840c-4b3f-818e-552865082e16"
+AZURE_AD_TOKEN_EXCHANGE_APP_ID="fb60f99c-7a34-4190-8149-302f77469936"
+# Create workload identity pool
```
Comment (saliceti) [inline scripts/azure-gcp-wif/create-gcp-workload-identity-pool-provider.sh:9]: Please insert blank line for more readability

File: documentation/azure-gcp-workload-identity-federation.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+Workload Identity Federation requires configuration on both Azure and GCP.
+
+## GCP Configuration
+
+To configure GCP run the gcloud scripts below.
+
+### Pre-requisites for running gcloud scripts
+
+- Install the gcloud cli for you operating system
```
Comment (saliceti) [inline documentation/azure-gcp-workload-identity-federation.md:13]: Can you link to https://cloud.google.com/sdk/docs/install

**Review by saliceti — APPROVED** [review]

**Pull request merged**